pyproject.toml

```diff
@@ -184,6 +184,71 @@ exclude = '''
 )
 '''
 
+[tool.ruff]
+line-length = 88
```

Comment:
Could also add

```
update-check = false  # there is a release every day
target-version = "py38"  # might only be important when we add ruff's pyupgrade features
fix = true  # compared to flake8/pylint, ruff has auto-rewrites to fix many rules
```

(btw. thank you for opening this PR - that's how I learned about ruff, I like it a lot!)

Reply:
are we sure we want `fix=true` here? because it might rewrite things which people don't want, such as 

https://github.com/pandas-dev/pandas/blob/2da7128e3436db520b96ccb07e56811a4bfcc72f/pandas/tests/frame/indexing/test_where.py#L824

(if there wasn't the `noqa`, then `--fix` would change `==False` to `is False`, though that's not the same in this context)

I think it'd be safer to let devs fix the issues? if you already have the other linting tools, then there's shouldn't be too many to deal with

Reply:
Given that ruff is still adding rules and removing false positives, it can make sense not to let it autofix. If it doesn't report issues (which should be the case on the CI), it also won't rewrite anything. Just thought that similar to black, it might be nice to also fix things :) I have no strong opinion on adding/not adding it.

Reply:
In theory black only reformats (the ast stays the same), and never actually rewrites. So that's less intrusive to let it autofix. 

I would also first get some experience with using ruff, we could always consider later to add autofixing.